Rakefile.rb

```diff
@@ -48,6 +48,36 @@ def name_from(json)
   orgs.first[:name]
 end
 
+def file_to_commit_metadata
+  # There's about 2MB of data returned by this command, so rather than
+  # using backticks to turning it into a string and then splitting it,
+  # parse the output line-by-line. As a bonus this doesn't
+  # unnecessarily invoke a shell.
+  command = ['git', '--no-pager', 'log', '--name-only',  '--format=%h|%at', '--', 'data']
+  result = {}
+  sha, timestamp = nil
+  Kernel.open('|-', 'r') { |f|
```

Comment:
This is a level of Ruby that I'm not entirely familiar with, but what's the advantage of dropping right down to this, rather than, say,
```ruby
    IO.popen(command) do |pipe|
      pipe.each_line do |line|
```
which seems to do the same thing, but without needing the extra level of `if f` indirection?